src/embed.ts

```diff
@@ -162,14 +162,14 @@ export abstract class Embed {
      * @param {models.ILoadConfiguration} config
      * @returns {Promise<void>}
      */
-    load(config: models.ILoadConfiguration, embedConfig?: IEmbedConfiguration): Promise<void> {
+    load(config: models.ILoadConfiguration): Promise<void> {
         const errors = models.validateLoad(config);
         if(errors) {
             throw errors;
         }
         
         let loadPath = '/report/load';
-        if(embedConfig && embedConfig.type === 'dashboard') {
+        if(config && (<IEmbedConfiguration>config).type === 'dashboard') {
```

Comment:
I'm not sure if I understand your proposal on the fix. How will the overridden 'load' methods find out the load path info? Through this.config:IInternalEmbedConfiguration in embed? Also wouldn't we have a lot of duplicated code in 3 load methods? All we want to change is the load path and I think the ideal place to differentiate is still in embed.

Technically you can't really prevent the user from doing crazy things such as embed a dashboard in a report iFrame...but it would just not work and user should be able to fix that relatively easily.

My proposal is to just fix it as:
if(this.config && this.config.type === 'dashboard') { ... }